crates/red_knot_python_semantic/resources/mdtest/annotations/annotated.md

```diff
@@ -0,0 +1,83 @@
+# `Annotated`
+
+`Annotated` attaches arbitrary metadata to a given type.
+
+## Usages
+
+`Annotated[T, ...]` is equivalent to `T`: All metadata arguments are simply ignored.
+
+```py
+from typing_extensions import Annotated
+
+def _(x: Annotated[int, "foo"]):
+    reveal_type(x)  # revealed: int
+
+def _(x: Annotated[int, lambda: 0 + 1 * 2 // 3, _(4)]):
+    reveal_type(x)  # revealed: int
+
+def _(
+    x: Annotated[
+        tuple[str, int, tuple[bytes]],
+        # TODO: This error is reporting the wrong thing
+        # error: [invalid-assignment]
+        (_ := b"SyntaxError: named expression cannot be used within an annotation"),
+    ]
+):
+    reveal_type(x)  # revealed: tuple[str, int, tuple[bytes]]
```

Comment:
you've added tests for very exotic kinds of metadata here, but I think we're still missing a simple test that's easy to read and understand, which demonstrates that we allow an arbitrary number of metadata elements without emitting a diagnostic or panicking. Just something like this:

```py
def _(x: Annotated[int, "arbitrary", "metadata", "elements", "are", "fine"]):
    reveal_type(x)  # revealed: int
```